app/js/trouble.js

```diff
@@ -53,15 +47,15 @@
 			$dropdown.data("status", 'closed');
 			$dropdown.removeClass('gbl_dropdown_active');
 			Array.prototype.foreach = function(cb) {
-				for (i = 0; i < this.length; ++i) {
+				for (var i = 0; i < this.length; ++i) {
 					cb(this[i]);
 				}
-			}
-			while (true) setTimeout(function() { setDropdownHeight(); }, 1000)
+			};
+			while (true) setTimeout(setDropdownHeight(), 1000);

```

Comment:
Yes, it would effectively be redundant, but what's actually going on here
is there's an anonymous function calling the setDropdownHeight function.
Since that function does nothing else, you could just replace the entire
thing with a reference to setDropdownHeight. If the idea was to potentially
add more code to that anonymous function expression later you would want to
maintain a function that was simply calling setDropdownHeight. A better
ultimate solution is simply to remove the while loop entirely (it's an
infinite loop) and just call setTimeout once.

thanks,

James

On Sat, Apr 11, 2015 at 12:30 PM, Curtis Nakauye <notifications@github.com>
wrote:

> In app/js/trouble.js
> <https://github.com/JSCRIPT300-Spring2015/Module_1/pull/18#discussion_r28197757>
> :
>
> >  					cb(this[i]);
> >  				}
> > -			}
> > -			while (true) setTimeout(function() { setDropdownHeight(); }, 1000)
> > +			};
> > +			while (true) setTimeout(setDropdownHeight(), 1000);
>
> looks like that setDropdownHeight function is already declared so moving
> it outside wouod be redundant no? can i just pull the parenthese so it
> doesn't execute right away?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JSCRIPT300-Spring2015/Module_1/pull/18/files#r28197757>
> .
>